src/main/java/org/cru/godtools/domain/packages/PageStructure.java

```diff
@@ -143,10 +164,178 @@ public void mergeXmlContent(Document updatedPageLayout)
+				if (!originalElementNodeName.equals(addElementNodeName) || !attrMatch)
+				{
+					String nodeToBeAddedString = XmlUtilities.xmlDocumentOrNodeToString(new DOMSource(nodeToAdd));
+
+					if (!currentXmlStringListOfElements.contains(nodeToBeAddedString))
+					{
+						Node targetNode = nodeToAdd.cloneNode(true);
+						Node nodeToBeImported = strippedDownCopyOfXmlContent.importNode(targetNode, true);
+						Node previousSibling = XmlUtilities.getPreviousSiblingElement(xmlContentNode);
+
+						if(previousSibling == null)
+						{
+							strippedDownCopyOfXmlContent.getDocumentElement().insertBefore(nodeToBeImported, xmlContentNode);
+						}
+						else
+						{
+							strippedDownCopyOfXmlContent.getDocumentElement().insertBefore(nodeToBeImported, previousSibling);
+						}
+					}
+				}
+			}
+			else
+			{
+				Node targetNode = strippedDownCopyOfXmlContent.importNode(nodeToAdd, true);
+				currentXmlContentNodeList.item(1).appendChild(targetNode);
+			}
+		}
+
+		xmlContent = strippedDownCopyOfXmlContent;
+	}
+
+	private boolean hasSameAttributes(Element oElement, Element aElement)
+	{
+		NamedNodeMap originalNamedNodeMap = oElement.getAttributes();
+		NamedNodeMap additionNamedNodeMap = aElement.getAttributes();
+
+		for (int n = 0; n < additionNamedNodeMap.getLength(); n++)
+		{
+			Attr a1 = (Attr) additionNamedNodeMap.item(n);
+			Attr o1 = (Attr) originalNamedNodeMap.item(n);
+
+			if(a1 == null ^ o1 ==null) return false;
+
+			if (!o1.getName().equals(a1.getName()) || !o1.getValue().equals(a1.getValue()))
+			{
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	public void removeXmlContent(Document documentWithRemovableElements) throws XMLStreamException, IOException,
+			ParserConfigurationException,SAXException,TransformerException
+	{
+		XmlUtilities.verifyDifferentXml(xmlContent, documentWithRemovableElements);
+
+		ByteArrayOutputStream originalXmlByteArrayOStream = XmlDocumentStreamConverter.writeToByteArrayStream(xmlContent);
+		ByteArrayOutputStream removableElementsByteArrayOStream = XmlDocumentStreamConverter.writeToByteArrayStream(documentWithRemovableElements);
+
+		XMLEventReader originalXmlReader = getXmlEventReaderFromByteArray(originalXmlByteArrayOStream);
+		XMLEventReader removableElementsXmlReader = getXmlEventReaderFromByteArray(removableElementsByteArrayOStream);
+
+		ByteArrayOutputStream byteArrayForDocument = new ByteArrayOutputStream();
+
+		XMLEventWriter xmlEventWriter = XMLOutputFactory.newInstance().createXMLEventWriter(byteArrayForDocument);
+		List<String> xmlEventArrayList = Lists.newArrayList();
+
+		boolean isFound = true ;
+		String deleteTagName = "";
+
+		//I'm sure there's a better way, but for now I write the results
+		//to a list and in the next while loop, we check if the event (xml element)
+		//exists.
+		while (removableElementsXmlReader.hasNext())
+		{
+			XMLEvent xmlEvent = removableElementsXmlReader.nextEvent();
+			xmlEventArrayList.add(xmlEvent.toString());
```

Comment:
I'm not sure this works.  It prints thinks like "Stax Event #5" which isn't guaranteed to be unique.  This leads to elements not being recognized as removed when they are actually removed.